module4/lessons/Week3/MaintainingState.md

```diff
@@ -0,0 +1,129 @@
+* In technical terms, what is a cookie?
+* Where are cookies stored?
+* How do you create a cookie in a .NET web application?
+
+Be ready to share out!
+
+</section>
+
+<section class='instructor-notes' markdown='1'>
+
+After giving students time to research, lead a discussion to answer these questions.  Make sure to hear from all voices in the room!  We want students to understand:
+* Cookies are _small_ pieces of data that are passed to servers from the client
+* Cookies are stored on instances of a client (this chrome browser, or this google user)
+* Cookies are typically created by a server
+* Cookies _can_ be created by a client, but not all servers will accept them or do anything with that info (dark mode).
+
+</section>
+
+### Inspecting Cookies
+
+Are you ever curious about what cookies are following you around?  With chrome dev tools, you can see the cookies that exist for a website that you are on!
+
+![Using Dev Tools to Show Cookies](/assets/images/module4/week3/InspectCookies.png)
+
+<section class='call-to-action' markdown='1'>
+
+**Try It!**
+In your browser, check out the cookies at some of your favorite websites.  How do these cookies change if you visit these sites in 'incognito' mode?
+
+</section>
+
+### Cookies in .NET
+
+To create a cookie in .NET we can add key/value pairs to the Response object.  All cookies are made of a key/value pair where the key and value are both strings (everything on the internet is strings!).
+
+<img src="/assets/images/module4/week3/AppendCookie.png" style="width: 50%;">
+
+This cookie will be set everytime a request is made:
+
+<img src="/assets/images/module4/week3/InspectCookies.png" style="width: 80%;">
+
+We can read cookies by asking or the values at specific keys from our Request object.
+
+<img src="/assets/images/module4/week3/ReadCookie.png" style="width: 50%;">
+
+<section class='instructor-notes' markdown='1'>
+
+The following activity could be completed individually, in groups, or a mix (start individually then switch to groups or vice-versa).  Ask students to share their solutions at the end.  Here is a possible solution:
+
+```c#
+        public IActionResult Index()
+        {
+            var visitCount = Request.Cookies["visitCount"];
+
+            if (visitCount is null)
+            {
+                Response.Cookies.Append("visitCount", "1");
+                ViewData["visitCount"] = 1;
+            }
+            else
+            {
+                visitCount = (Convert.ToInt16(visitCount) + 1).ToString();
+                Response.Cookies.Append("visitCount", visitCount);
+                ViewData["visitCount"] = visitCount;
+            }
+
+            return View();
+        }
+```
+
+</section>
+
+### Implement a Cookie
+
+<section class='call-to-action' markdown='1'>
+
+Let's try it out.  Fork and clone this [starter repo](https://github.com/turingschool-examples/GoodBooksMvcStarter)
```

Comment:
I like this activity a lot! Definitely has a bit of logic for students to work through.

I wonder if somewhere we should say that the browser will automatically send all of the cookies that belong to that domain. No work in the view is required!

Reply:
I think I'll add this as an instructor note - it would be great if students 'discover' this a little bit (or feel like they do)